iroh/src/node.rs

```diff
@@ -173,8 +171,11 @@ impl<D: BaoStore> Node<D> {
     pub async fn shutdown(self) -> Result<()> {
         self.inner.cancel_token.cancel();
 
-        if let Ok(task) = Arc::try_unwrap(self.task) {
-            task.await?;
+        if let Ok(mut task) = Arc::try_unwrap(self.task) {
+            task.take().expect("cannot be empty").await?;
+        }
+        if let Some(mut tasks) = self.inner.tasks.lock().unwrap().take() {
```

Comment:
you could skip the option I believe

Reply:
did that and also moved into `NodeInner`